python/pyarrow/tests/test_compute.py

```diff
@@ -2223,6 +2223,29 @@ def _check_datetime_components(timestamps, timezone=None):
         first_week_is_fully_in_year=False)
     assert pc.week(tsa, options=week_options).equals(pa.array(iso_week))
 
+    # Test for iso_calendar regression
```

Comment:
Can you move this to a separate test? The `_check_datetime_components` function is already quite long, and the below also doesn't actually use `timestamps`, so it's a bit strange to put it here)

Reply:
Fair point! Moved to `_test_iso_calendar_regression`.